crowbar_framework/app/controllers/application_controller.rb

```diff
@@ -33,6 +33,9 @@ class ApplicationController < ActionController::Base
     Crowbar::Installer.successful? || \
     Rails.env.test?
   }
+  before_filter :upgrade, if: proc {
+    File.exist?("/var/lib/crowbar/upgrade/progress.yml")
```

Comment:
@skazi0 it should be as cheap as possible, as we are running this check on every request. there might be even better ways, but a check for file existence seems quicker than a full parse of a status file. 
i thought about `Rails.cache` too which would be really quick I guess. It could be written at the time of the prechecks and after finishing the upgrade. The only problem is that it won't persist after a reboot.